### PR TITLE
Adds support for running integration tests against Airflow

### PR DESCRIPTION
**Pull request by saurav-c**
## Describe your changes and why you are making these changes
This PR adds initial support for running integration tests against Airflow. It has a setup script for starting a local Airflow instance. It also adds logic for manually copying the Airflow DAG file to the appropriate folder.

There is still follow on work that needs to be done before these tests can be run on Github actions:
- Fix syncing bugs https://linear.app/aqueducthq/issue/ENG-2794/fix-bug-where-airflow-dags-created-by-integration-test-are-not-syncing
- Configure running these tests on github actions https://linear.app/aqueducthq/issue/ENG-2796/configure-airflow-tests-to-run-on-github-actions

## Related issue number (if any)
ENG 2116

## Tests
Ran the basic flow_test locally and uncovered the syncing bug.

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: integration_tests/sdk/shared/flow_helpers.py [hunk 1]
```diff
@@ -82,6 +88,23 @@ def publish_flow_test(
...
+        dag_path = f"{name}_airflow.py"
+        assert os.path.exists(
+            dag_path
+        ), f"The expected Airflow DAG file was not found at {dag_path}"
+
+        folder_path = os.path.expanduser(AIRFLOW_DAGS_FOLDER)
+        shutil.move(dag_path, folder_path)
+
+        print("Sleeping for 20s to wait for Airflow scheduler to pick up new DAG file")
+        time.sleep(20)
```
Comment (kenxu95) [inline integration_tests/sdk/shared/flow_helpers.py:103]: Is this sleeping guaranteed to work? What happens if the scheduler doesn't pick it up in time? Does trigger fail or something.

Comment (saurav-c) [reply integration_tests/sdk/shared/flow_helpers.py:103]: We configure the scheduler to check for new DAGs every 15 seconds so this should be ok. We can bump it to 30s to be safe. Yes, trigger will fail if the scheduler doesn't pick up the new DAG.

Comment (saurav-c) [reply integration_tests/sdk/shared/flow_helpers.py:103]: Configured here https://github.com/aqueducthq/aqueduct/pull/1204/files#diff-fd66f15b393a531842dfb747a1dd0c1cdc7ddbd210a4daafbda5ac23a3021d76R15

File: integration_tests/sdk/shared/flow_helpers.py [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
 from aqueduct.artifacts.base_artifact import BaseArtifact
-from aqueduct.constants.enums import ExecutionStatus
+from aqueduct.constants.enums import ExecutionStatus, ServiceType
 
 import aqueduct
 from aqueduct import Flow
 
+from .compute import type_from_engine_name
+
+AIRFLOW_DAGS_FOLDER = "~/airflow/dags"
```
[the remark this answers is not in the document]
Comment (saurav-c) [reply integration_tests/sdk/shared/flow_helpers.py:15]: It should be guaranteed to exist, but also added a safety check as part of the Airflow setup script here: https://github.com/aqueducthq/aqueduct/pull/1204/files#diff-fd66f15b393a531842dfb747a1dd0c1cdc7ddbd210a4daafbda5ac23a3021d76R18

[the remark this answers is not in the document]
Comment (kenxu95) [reply integration_tests/sdk/shared/flow_helpers.py:15]: ok, can you note that here right above the constant?

File: integration_tests/sdk/shared/flow_helpers.py [hunk 3]
```diff
@@ -211,6 +234,12 @@ def delete_all_flows(client: aqueduct.Client) -> None:
         else:
             print("Successfully deleted workflow %s" % flow_name)
 
+        # Try deleting Airflow DAG file if it exists
+        # TODO - Only do this for Airflow workflows
```
Comment (kenxu95) [inline integration_tests/sdk/shared/flow_helpers.py:238]: Should we make a task for this TODO?

**Review by kenxu95 — APPROVED** [review]
Cool, just a couple tiny nits.

**Pull request merged**